File: front-end/src/types/ExerciseStatisticsDTO.ts
export interface ExerciseStatisticsDTO {
  exerciseTitle: string;
  totalStudents: number;
  totalAttempts: number;
  averageScore: number;
  averagePercentage: number;
  passedCount: number;
  failedCount: number;
  passRate?: number;
  averageTimeSpent: number;
  highestScore: number;
  lowestScore: number;
}

File: back-end/src/main/java/com/meutcc/backend/content/exercise/dtos/ExerciseStatisticsDTO.java
package com.meutcc.backend.content.exercise.dtos;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.math.BigDecimal;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ExerciseStatisticsDTO(
        Integer totalStudents,
        Integer totalAttempts,
        BigDecimal averageScore,
        BigDecimal averagePercentage,
        Integer passedCount,
        Integer failedCount,
        Double passRate,
        Integer averageTimeSpent,
        BigDecimal highestScore,
        BigDecimal lowestScore
) {
}